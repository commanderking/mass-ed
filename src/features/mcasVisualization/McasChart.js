import React from 'react';
import { VictoryChart, VictoryLabel, VictoryGroup, VictoryBar } from 'victory';
import { graphColors } from './mcasConstants';

const formatDataForChart = (school, index) => {
  return categories.map(category => {
    return {
      x: category,
      y: school[mapCategoriesToRawDataValue[category]],
      fill: graphColors[index]
    }
  });
}

const categories = ['Exceeding', 'Meeting', 'Partially Meeting', 'Not Meeting'];

const mapCategoriesToRawDataValue = {
  [categories[0]] : 'exceededPercent',
  [categories[1]] : 'metPercent',
  [categories[2]] : 'partiallyMetPercent',
  [categories[3]] : 'notMetPercent'
};

export const graphConstants = {
  DOMAIN_PADDING: 40,
  TITLE_X: 225,
  TITLE_Y: 20,
  X_AXIS_LABEL_PADDING: 30,
  Y_AXIS_LABEL_PADDING: 40,
  Y_AXIS_TEXT: '% Students',
  BAR_FILL: '#c43a31',
  BAR_STROKE: 'black',
  BAR_STROKE_WIDTH: 2
}

const McasChart = ({ selectedSchools }) => {
  return (
    <div className='mcasChartWrapper'>
      <VictoryChart
        domainPadding={graphConstants.DOMAIN_PADDING}
      >
        <VictoryLabel
          text={'MCAS 2017 Scores'}
          textAnchor='middle'
          x={250}
          y={20}
        />
        <VictoryGroup
          offset={20}
          padding={-50}
          colorScale={"qualitative"}
          categories={{ x: categories}}
        >
          {
            selectedSchools.map((school, index) => {
              return (
                <VictoryBar
                  key={`${school.name}-${school.schoolCode}`}
                  labels={(d) => `${d.y}%`}
                  style={{
                    data: {
                      width: 18,
                      padding: 5
                    },
                    labels: {
                      fontSize: 10
                    }
                  }}
                  data={formatDataForChart(school, index)}
                />
              )
            })
          }
        </VictoryGroup>
      </VictoryChart>
    </div>
  )

}

export { McasChart };
