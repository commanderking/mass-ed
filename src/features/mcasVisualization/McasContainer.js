// flow
import React from 'react';
import { connect } from 'react-redux';

import { VictoryChart, VictoryBar, VictoryAxis, VictoryLabel } from 'victory';
import { addSchoolAction, selectSchoolAction } from './mcasActions';
import { mcasData } from './mcasData';
import VirtualizedSelect from 'react-virtualized-select';

// This only needs to be done once; probably during bootstrapping process.
import 'react-select/dist/react-select.css'
import 'react-virtualized/styles.css'
import 'react-virtualized-select/styles.css'

const categories = ['Exceeding', 'Meeting', 'Partially Meeting', 'Not Meeting'];

const mcasDataConstants = {
  SCHOOL_NAME: 'School Name'
}

export const graphConstants = {
  DOMAIN_PADDING: 40,
  TITLE_X: 225,
  TITLE_Y: 20,
  X_AXIS_LABEL_PADDING: 30,
  Y_AXIS_LABEL_PADDING: 40,
  Y_AXIS_TEXT: '% Students',
  BAR_FILL: '#c43a31',
  BAR_STROKE: 'black',
  BAR_STROKE_WIDTH: 2
}

const mapCategoriesToRawDataValue = {
  [categories[0]] : 'E %',
  [categories[1]] : 'M %',
  [categories[2]] : 'PM %',
  [categories[3]] : 'NM %'
};

const formatDataForChart = school => {
  return categories.map(category => {
    return { x: category, y: school[mapCategoriesToRawDataValue[category]]}
  })
}

const parseSchoolName = (schoolNameWithDistrict) => {
  return schoolNameWithDistrict.split('-')[1].trim()
}

const getSchoolNames = (allSchools) => {
  const data = allSchools.map((school, index) => {
    return {
      value: school[mcasDataConstants.SCHOOL_NAME],
      label: school[mcasDataConstants.SCHOOL_NAME],
      index: index
    }
  });
  return data;
}

export const UnwrappedMcasContainer = ({ allSchools, selectedSchoolIndexes, dropdownSchoolIndex, addSchoolClick, selectSchool }) => (
  <div>
    <div className="schoolSelectWrapper">
      <VirtualizedSelect
        options={getSchoolNames(allSchools)}
        onChange={(selectValue) => {
          if (selectValue) {
            selectSchool(selectValue.index);
          }
        }}
        value={allSchools[dropdownSchoolIndex][mcasDataConstants.SCHOOL_NAME]}
      />
      <button onClick={ () => {
        console.log('school selected');
        addSchoolClick(dropdownSchoolIndex);
      }}>Add School</button>
    </div>
    {selectedSchoolIndexes.map((schoolIndex) => {
      const schoolData = allSchools[schoolIndex];
      return (
        <div key={schoolData[mcasDataConstants.SCHOOL_NAME]} className='chartWrapper'>
          <VictoryChart domainPadding={graphConstants.DOMAIN_PADDING} >
            <VictoryLabel
              text={parseSchoolName(schoolData[mcasDataConstants.SCHOOL_NAME])}
              textAnchor="middle"
              x={graphConstants.TITLE_X}
              y={graphConstants.TITLE_Y}
            />
            <VictoryBar
              data={formatDataForChart(schoolData)}
              categories={{ x: categories}}
              labels={(d) => `${d.y}%`}
              style={{
                data: {
                  fill: graphConstants.BAR_FILL,
                  stroke: graphConstants.BAR_STROKE,
                  strokeWidth: graphConstants.BAR_STROKE_WIDTH
                }
              }}
              />
            <VictoryAxis
              style={{
                axisLabel: { padding: graphConstants.X_AXIS_LABEL_PADDING }
              }}
            />
            <VictoryAxis dependentAxis
              label={graphConstants.Y_AXIS_TEXT}
              style={{
                axisLabel: { padding: graphConstants.Y_AXIS_LABEL_PADDING }
              }}
              tickFormat={(t) => `${Math.round(t)}%`}
            />
          </VictoryChart>
          <button>x</button>
        </div>
      )
    })}

  </div>
);

const mapStateToProps = state => {
  const { selectedSchoolIndexes, dropdownSchoolIndex } = state;
  return {
    allSchools: mcasData, // Hard coded for now, move to entity eventually
    selectedSchoolIndexes,
    dropdownSchoolIndex
  };
}

const mapDispatchToProps = dispatch => {
  return {
    addSchoolClick: schoolData => {
      console.log('in addSchoolClick');
      dispatch(addSchoolAction(schoolData));
    },
    selectSchool: schoolIndex => {
      dispatch(selectSchoolAction(schoolIndex))
    }
  }
}

const McasContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(UnwrappedMcasContainer);

export { McasContainer };
