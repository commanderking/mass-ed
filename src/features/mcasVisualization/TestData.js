export const allSchoolsMockData = [{
  "School Name": "Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School",
  "School Code": 4450105,
  "Subject": "ELA",
  "M+E #": 307,
  "M+E %": 42,
  "E #": 23,
  "E %": 3,
  "M #": 284,
  "M %": 39,
  "PM #": 362,
  "PM %": 50,
  "NM #": 55,
  "NM %": 8,
  "Student Included": 724,
  "Avg. Scaled Score": 496.2,
  "SGP": 55,
  "Included In SGP": 587,
  "Ach. PCTL": 36
},
{
  "School Name": "Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School",
  "School Code": 4450105,
  "Subject": "MATH",
  "M+E #": 255,
  "M+E %": 35,
  "E #": 16,
  "E %": 2,
  "M #": 239,
  "M %": 33,
  "PM #": 405,
  "PM %": 56,
  "NM #": 64,
  "NM %": 9,
  "Student Included": 724,
  "Avg. Scaled Score": 494,
  "SGP": 48,
  "Included In SGP": 587,
  "Ach. PCTL": 29
}];
