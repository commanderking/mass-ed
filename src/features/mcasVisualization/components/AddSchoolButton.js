import React from "react";
import { Button } from "reactstrap";

const AddSchoolButton = ({ schoolGroup, onClick, disabled }) => (
  <Button disabled={disabled} onClick={onClick} color="primary">
    ADD {schoolGroup}
  </Button>
);

export { AddSchoolButton };
