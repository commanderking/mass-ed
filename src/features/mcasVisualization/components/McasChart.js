// @flow
import React from "react";
import { VictoryChart, VictoryLabel, VictoryGroup, VictoryBar } from "victory";
import { graphColors } from "../mcasConstants";

import type { schoolMcasType } from "./mcas.flow";

type schoolDataForChart = {
  x: "Exceeding" | "Meeting" | "Partially Meeting" | "Not Meeting",
  y: number,
  fill: string
};

const formatDataForChart = (
  school: schoolMcasType,
  index: number
): Array<schoolDataForChart> => {
  return categories.map(category => {
    return {
      x: category,
      y: school[mapCategoriesToRawDataValue[category]],
      fill: graphColors[index]
    };
  });
};

const categories = ["Exceeding", "Meeting", "Partially Meeting", "Not Meeting"];

const mapCategoriesToRawDataValue = {
  [categories[0]]: "exceededPercent",
  [categories[1]]: "metPercent",
  [categories[2]]: "partiallyMetPercent",
  [categories[3]]: "notMetPercent"
};

export const graphConstants = {
  DOMAIN_PADDING: 40
};

type Props = {
  selectedSchools: Array<schoolMcasType>
};

const McasChart = ({ selectedSchools }: Props) => {
  return (
    <div className="mcasChartWrapper">
      <VictoryChart domainPadding={graphConstants.DOMAIN_PADDING}>
        <VictoryLabel
          text={"MCAS 2017 Scores"}
          textAnchor="middle"
          x={250}
          y={20}
        />
        <VictoryGroup
          offset={20}
          padding={-50}
          colorScale={"qualitative"}
          categories={{ x: categories }}
        >
          {selectedSchools.map((school, index) => {
            return (
              <VictoryBar
                key={`${school.schoolName}-${school.schoolCode}`}
                labels={d => `${d.y}%`}
                style={{
                  data: {
                    width: 18,
                    padding: 5
                  },
                  labels: {
                    fontSize: 10
                  }
                }}
                data={formatDataForChart(school, index)}
              />
            );
          })}
        </VictoryGroup>
      </VictoryChart>
    </div>
  );
};

export { McasChart };
