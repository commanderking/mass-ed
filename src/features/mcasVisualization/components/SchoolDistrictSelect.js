import React from "react";
import "rc-select/assets/index.css";
import Select, { Option } from "rc-select";
import { parseSchoolNameFromCompleteName } from "../../../common/utils/formatUtils";
import type { SchoolMcasType } from "./mcas.flow.js";

type Props = {
  selectSchool: SchoolCodeType => void,
  allSchools: Array<SchoolMcasType>,
  disabled: boolean
};

const SchoolDistrictSelect = ({
  selectSchool,
  allSchools,
  disabled
}: Props) => {
  return (
    <div style={{ width: 300 }}>
      <Select
        id="SelectId"
        style={{ width: 500 }}
        onSelect={(selectValue, option) => {
          if (selectValue) {
            selectSchool(option.props.code);
          }
        }}
        dropdownStyle={{
          height: "300px",
          overflow: "scroll"
        }}
        disabled={disabled}
        allowClear
        combobox
      >
        {allSchools.map((school, index) => {
          // TODO: Selector should really handle any parsing before we get into the component
          const name = parseSchoolNameFromCompleteName(school.name);
          return (
            <Option
              key={index.toString()}
              value={name}
              index={index}
              code={school.code}
            >
              {name}
            </Option>
          );
        })}
      </Select>
    </div>
  );
};

export { SchoolDistrictSelect };
