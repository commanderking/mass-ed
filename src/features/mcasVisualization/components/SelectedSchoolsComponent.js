import React from "react";
import { parseSchoolNameFromCompleteName } from "../../../common/utils/formatUtils";
import { SchoolLabel } from "./SchoolLabel";

const SelectedSchoolsComponent = ({ selectedSchools, deleteSchool }) => (
  <div id="SelectedSchoolsComponent" className="schoolLabelsWrapper">
    <h3>Selected Schools</h3>
    {selectedSchools.map((school, index) => {
      const { name, code } = school;
      return (
        <SchoolLabel
          key={name}
          index={index}
          name={parseSchoolNameFromCompleteName(name)}
          code={code}
          deleteSchool={deleteSchool}
        />
      );
    })}
  </div>
);

export { SelectedSchoolsComponent };
