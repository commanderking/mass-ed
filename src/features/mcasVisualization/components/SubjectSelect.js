import React from "react";
import { Button, ButtonGroup } from "reactstrap";
import { subjectsConstants } from "../mcasConstants";
export const subjects = [subjectsConstants.ELA, subjectsConstants.MATH];

const isSelectedSubject = (subject, selectedSubject) =>
  subject === selectedSubject;

const formatBackgroundColorForSubject = (subject, selectedSubject) =>
  isSelectedSubject(subject, selectedSubject) ? "#006CD6" : "gray";

const renderSubjectButtons = (setSubject, selectedSubject) =>
  subjects.map(subject => (
    <Button
      key={`subjectSelect-${subject}-button`}
      style={{
        backgroundColor: formatBackgroundColorForSubject(
          subject,
          selectedSubject
        )
      }}
      onClick={() => {
        setSubject(subject);
      }}
    >
      {subject}
    </Button>
  ));

const SubjectSelect = ({ setSubject, selectedSubject }) => (
  <ButtonGroup className="subjectSelect">
    {renderSubjectButtons(setSubject, selectedSubject)}
  </ButtonGroup>
);

export { SubjectSelect };
