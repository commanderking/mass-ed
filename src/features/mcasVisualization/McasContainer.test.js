import React from "react";
import { allSchoolsMockData } from "./TestData";
import { UnwrappedMcasContainer } from "./McasContainer";
import { shallow } from "enzyme";
import { McasChart } from "./components/McasChart";
import { SubjectSelect } from "./components/SubjectSelect";
import { SchoolDistrictSelect } from "./components/SchoolDistrictSelect";
import { AddSchoolButton } from "./components/AddSchoolButton";
import { SelectedSchoolsComponent } from "./components/SelectedSchoolsComponent";

let mockProps;

describe("McasContainer", () => {
  beforeEach(() => {
    mockProps = {
      allSchoolsBySubject: allSchoolsMockData,
      addAllSchools: jest.fn(),
      selectedSchools: [],
      dropdownCode: 0,
      addSchoolClick: jest.fn(),
      selectSchool: jest.fn(),
      deleteSchool: jest.fn()
    };
  });

  it("renders with correct structure if no school selected", () => {
    const wrapper = shallow(<UnwrappedMcasContainer {...mockProps} />);
    expect(wrapper.find(SchoolDistrictSelect)).toHaveLength(1);
    expect(wrapper.find(McasChart)).toHaveLength(0);
    expect(wrapper.find(SelectedSchoolsComponent)).toHaveLength(0);
  });

  it("renders with correct structure if no school selected", () => {
    mockProps.selectedSchools = [
      { code: 22, name: "Arlington", subject: "ELA" }
    ];
    const wrapper = shallow(<UnwrappedMcasContainer {...mockProps} />);
    expect(wrapper.find(SchoolDistrictSelect)).toHaveLength(1);
    expect(wrapper.find(McasChart)).toHaveLength(1);
    expect(wrapper.find(SelectedSchoolsComponent)).toHaveLength(1);
  });

  it("clicking add school button fires correct action", () => {
    const wrapper = shallow(<UnwrappedMcasContainer {...mockProps} />);
    const buttonWrapper = wrapper.find(AddSchoolButton);
    expect(buttonWrapper.props()).toHaveProperty("disabled", false);
    buttonWrapper.simulate("click");
    expect(mockProps.addSchoolClick).toHaveBeenCalledTimes(1);
    expect(mockProps.addSchoolClick).toHaveBeenCalledWith(
      mockProps.allSchoolsBySubject[0].code
    );
  });

  it("disables add school button when four schools are set", () => {
    mockProps.selectedSchools = allSchoolsMockData;
    const wrapper = shallow(<UnwrappedMcasContainer {...mockProps} />);
    const buttonWrapper = wrapper.find(AddSchoolButton);
    expect(buttonWrapper.props()).toHaveProperty("disabled", true);
  });

  it("disables add school button when four schools are set", () => {
    mockProps.selectedSchools = allSchoolsMockData;
    const wrapper = shallow(<UnwrappedMcasContainer {...mockProps} />);
    const schoolSelectWrapper = wrapper.find(SchoolDistrictSelect);
    expect(schoolSelectWrapper.props()).toHaveProperty("disabled", true);
  });
});
