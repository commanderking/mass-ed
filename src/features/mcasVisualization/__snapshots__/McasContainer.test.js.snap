// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`McasContainer renders graph components renders correct bar component 1`] = `
<VictoryBar
  categories={
    Object {
      "x": Array [
        "Exceeding",
        "Meeting",
        "Partially Meeting",
        "Not Meeting",
      ],
    }
  }
  containerComponent={
    <VictoryContainer
      className="VictoryContainer"
      portalComponent={<Portal />}
      portalZIndex={99}
      responsive={true}
    />
  }
  data={
    Array [
      Object {
        "x": "Exceeding",
        "y": 3,
      },
      Object {
        "x": "Meeting",
        "y": 39,
      },
      Object {
        "x": "Partially Meeting",
        "y": 50,
      },
      Object {
        "x": "Not Meeting",
        "y": 8,
      },
    ]
  }
  dataComponent={<Bar />}
  groupComponent={
    <g
      role="presentation"
    />
  }
  labelComponent={
    <VictoryLabel
      capHeight={0.71}
      lineHeight={1}
    />
  }
  labels={[Function]}
  scale="linear"
  standalone={true}
  style={
    Object {
      "data": Object {
        "fill": "#c43a31",
        "stroke": "black",
        "strokeWidth": 2,
      },
    }
  }
  theme={
    Object {
      "area": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "axis": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "axis": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": 1,
          },
          "axisLabel": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 25,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
          "grid": Object {
            "fill": "none",
            "pointerEvents": "visible",
            "stroke": "none",
          },
          "tickLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "ticks": Object {
            "fill": "transparent",
            "size": 1,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "bar": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
            "padding": 8,
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "candlestick": Object {
        "candleColors": Object {
          "negative": "#252525",
          "positive": "#ffffff",
        },
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "chart": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "errorbar": Object {
        "borderWidth": 8,
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "group": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "legend": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "gutter": 10,
        "orientation": "vertical",
        "style": Object {
          "data": Object {
            "type": "circle",
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "title": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 5,
            "stroke": "transparent",
          },
        },
        "titleOrientation": "top",
      },
      "line": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "pie": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 400,
        "padding": 50,
        "style": Object {
          "data": Object {
            "padding": 10,
            "stroke": "transparent",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 20,
            "stroke": "transparent",
          },
        },
        "width": 400,
      },
      "scatter": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "stack": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "tooltip": Object {
        "cornerRadius": 5,
        "flyoutStyle": Object {
          "fill": "#f0f0f0",
          "pointerEvents": "none",
          "stroke": "#252525",
          "strokeWidth": 1,
        },
        "pointerLength": 10,
        "style": Object {
          "fill": "#252525",
          "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
          "fontSize": 14,
          "letterSpacing": "normal",
          "padding": 5,
          "pointerEvents": "none",
          "stroke": "transparent",
          "textAnchor": "middle",
        },
      },
      "voronoi": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "flyout": Object {
            "fill": "#f0f0f0",
            "pointerEvents": "none",
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 5,
            "pointerEvents": "none",
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
    }
  }
/>
`;

exports[`McasContainer renders graph components renders correct graph title 1`] = `
<VictoryLabel
  capHeight={0.71}
  lineHeight={1}
  text="Abby Kelley Foster Charter Public School"
  textAnchor="middle"
  x={225}
  y={20}
/>
`;

exports[`McasContainer renders with correct structure 1`] = `
<div>
  <div
    className="schoolSelectWrapper"
  >
    <VirtualizedSelect
      async={false}
      maxHeight={200}
      onChange={[Function]}
      optionHeight={35}
      options={
        Array [
          Object {
            "index": 0,
            "label": "Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School",
            "value": "Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School",
          },
          Object {
            "index": 1,
            "label": "Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School",
            "value": "Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School",
          },
        ]
      }
      value="Abby Kelley Foster Charter Public (District) - Abby Kelley Foster Charter Public School"
    />
    <button
      onClick={[Function]}
    >
      Add School
    </button>
  </div>
</div>
`;
